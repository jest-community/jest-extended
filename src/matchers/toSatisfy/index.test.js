import matcher from './';

expect.extend(matcher);

describe('.toSatisfy', () => {
  const is2 = n => n === 2;

  it('passes when given a function that returns true', () => {
    expect(2).toSatisfy(is2);
  });

  it('fails when given function that returns false', () => {
    expect(() => expect(3).toSatisfy(is2)).toThrowErrorMatchingSnapshot();
  });
});

describe('.not.toSatisfy', () => {
  const isTrue = a => a === true;

  it('passes when given a function that returns false', () => {
    expect(false).not.toSatisfy(isTrue);
  });

  it('fails when given a function that returns true', () => {
    expect(() => expect(true).not.toSatisfy(isTrue)).toThrowErrorMatchingSnapshot();
  });
});
