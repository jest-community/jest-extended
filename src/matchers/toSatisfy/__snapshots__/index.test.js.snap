// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.not.toSatisfy fails when given a function that returns true 1`] = `
"<dim>expect(</><red>received</><dim>).not.toSatisfy()</>

Expected value to not satisfy:
  <green>[Function isTrue]</>
Received:
  <red>true</>"
`;

exports[`.toSatisfy fails when given function that returns false 1`] = `
"<dim>expect(</><red>received</><dim>).toSatisfy()</>

Expected value to satisfy:
  <green>[Function is2]</>
Received:
  <red>3</>"
`;
